test/export/test_export.py

```diff
@@ -4396,6 +4421,7 @@ def forward(self):
         ep = export(m, ())
         self.assertEqual(ep.graph_signature.lifted_tensor_constants, ["x"])
 
+    @unittest.expectedFailure
```

Comment:
This PR seems to expose some already existing problem between explicitly static constraints + dataclasses (or equality constraints)? Marking this as a failure to open the rest of the PR for review, but will work on this and resolve it before this PR is merged in.

Reply:
OK, as long as you get to it. I'd guess info on unused inputs could easily be perturbed.

Reply:
about to add a fix in the next commit